### PR TITLE
Hds 1816 header custom theming

### PR DESCRIPTION
**Pull request by minnas**
## Description

This PR adds support for header custom theming which gives possibility to use custom brand colours in header.

## Motivation and Context

[HDS-1816](https://helsinkisolutionoffice.atlassian.net/browse/HDS-1816)

## How Has This Been Tested?

Tested Locally 
(Browsers: Firefox, Chrome, Safari)

## Demos

[Storybook](https://city-of-helsinki.github.io/hds-demo/header-custom-theme/?path=/story/components-header--with-full-features-custom-theme)
[Site](https://city-of-helsinki.github.io/hds-demo/header-custom-theme-site/components/header/)

## Screenshot:

header-custom-theme

[HDS-1816]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-1816?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: site/src/docs/components/header/index.mdx [hunk 1]
```diff
@@ -106,5 +116,21 @@ Coming soon!
         <Header.NavigationLink href="#" label="Link 3" />
       </Header.NavigationMenu>
     </Header>
+<<<<<<< HEAD
```
Comment (mrTuomoK) [inline site/src/docs/components/header/index.mdx:119]: Erh... some rebasing leftovers here -> remove 😄

Comment (minnas) [reply site/src/docs/components/header/index.mdx]: fixed now and rebased to `hds-1803-header-dark-theme-support`

**Comment by NikoHelle** [comment]
The demo has same issue the dark theme had: focus is invisible in the action bar (logo, city name, languages).

**Comment by timwessman** [comment]
Good work! I'll say this is ready to merge after that focus bug is fixed

**Comment by minnas** [comment]
I close this one and create new PR for header custom theming

**Pull request closed without merging**